src/main/java/pl/edu/agh/ki/io/security/UserPrincipal.java

```diff
@@ -0,0 +1,74 @@
+package pl.edu.agh.ki.io.security;
+
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.oauth2.core.user.OAuth2User;
+import pl.edu.agh.ki.io.models.User;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Map;
+
+@NoArgsConstructor
+@Getter
+public class UserPrincipal implements UserDetails, OAuth2User {
+    private User user;
+    private Map<String, Object> attributes;
+
+    public UserPrincipal(User user, Map<String, Object> attributes) {
+        this.user = user;
+        this.attributes = attributes;
+    }
+
+    public UserPrincipal(User user) {
+        this.user = user;
+    }
+
+    @Override
+    public Map<String, Object> getAttributes() {
+        return this.attributes;
+    }
```

Comment:
Czy to jest potrzebne jesli jest annotacja `@Getter` nad klasa? Mozesz sprobowac wywalic i zobaczyc czy dalej dziala

Reply:
wow, rzeczywiście, myślałem że samo nadpisywanie ma znaczenie, ale okazuje się że nie